src/Tests/Topology/When_configuring_publishers.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace NServiceBus.Azure.WindowsAzureServiceBus.Tests.Topology
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using NServiceBus.AzureServiceBus;
+    using NUnit.Framework;
+
+    [TestFixture]
+    [Category("AzureServiceBus")]
+    public class When_configuring_publishers
+    {
+        private PublishersConfiguration _configuration;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var conventions = new FakeConventions();
+            _configuration = new PublishersConfiguration(conventions);
+        }
+
+        [Test]
+        public void Should_throw_if_mapping_does_not_exist()
+        {
+            Assert.Throws<KeyNotFoundException>(() => _configuration.GetPublishersFor(typeof(MyType)));
```

Comment:
Should it throw a generic `KeyNotFoundException` or something that will contain more information to help to configure mappings properly?